### PR TITLE
tools/scylla-sstable: load_sstables: handle load errors

### LINKED ISSUE #15359: Unhandled error when scylla-sstable fails to dump non-cdc sstables due to different partitioner
This looks similar to #9840, but it is the exact opposite.
scylla-sstable fails with the following error:
```
ERROR 2023-09-12 08:02:58,124 [shard 0:main] seastar - Exiting on unhandled exception: std::runtime_error (SSTable /home/bhalevy/.dtest/dtest-dxg4xdxg/test/node1/data/ks/cf-a3009f20512911ee8000d81cd2da3fd7/me-3g9b_0e0x_39vtt1y2rcqrffz55j-big-Data.db uses org.apache.cassandra.dht.Murmur3Partitioner partitioner which is different than com.scylladb.dht.CDCPartitioner partitioner used by the database)
```
where a regular sstable uses the `Murmur3Partitioner` while the tool claims the `CDCPartitioner` is used by the database.

### PR DESCRIPTION
**Pull request by bhalevy**
Currently, exceptions thrown from `sst->load` are unhandled, resulting in, e.g.:
```
ERROR 2023-09-12 08:02:58,124 [shard 0:main] seastar - Exiting on unhandled exception: std::runtime_error (SSTable /home/bhalevy/.dtest/dtest-dxg4xdxg/test/node1/data/ks/cf-a3009f20512911ee8000d81cd2da3fd7/me-3g9b_0e0x_39vtt1y2rcqrffz55j-big-Data.db uses org.apache.cassandra.dht.Murmur3Partitioner partitioner which is different than com.scylladb.dht.CDCPartitioner partitioner used by the database)
```

Log the errors and exit the tool with non-zero status in this case.

Fixes #15359

### CODE REVIEW

File: tools/scylla-sstable.cc [hunk 1]
```diff
@@ -258,7 +258,12 @@ const std::vector<sstables::shared_sstable> load_sstables(schema_ptr schema, sst
         data_dictionary::storage_options local;
         auto sst = sst_man.make_sstable(schema, dir_path.c_str(), local, ed.generation, sstables::sstable_state::normal, ed.version, ed.format);
 
-        co_await sst->load(schema->get_sharder(), sstables::sstable_open_config{.load_first_and_last_position_metadata = false});
+        try {
+            co_await sst->load(schema->get_sharder(), sstables::sstable_open_config{.load_first_and_last_position_metadata = false});
+        } catch (...) {
+            sst_log.error("Could not load SSTable: {}: {}", sst->get_filename(), std::current_exception());
```
Comment (denesb) [inline tools/scylla-sstable.cc:264]: Let's use a nested exception, with the extra context here and...

File: tools/scylla-sstable.cc [hunk 2]
```diff
@@ -2889,7 +2894,11 @@ Validate the specified sstables:
 
         std::vector<sstables::shared_sstable> sstables;
         if (app_config.count("sstables")) {
-            sstables = load_sstables(schema, sst_man, app_config["sstables"].as<std::vector<sstring>>());
+            try {
+                sstables = load_sstables(schema, sst_man, app_config["sstables"].as<std::vector<sstring>>());
+            } catch (...) {
+                return 1;
```
Comment (denesb) [inline tools/scylla-sstable.cc:2900]: ... print all exceptions here. There can be exceptions from other parts, not just load() and we will now silently swallow those.

**Review by denesb — CHANGES_REQUESTED** [review]

File: tools/scylla-sstable.cc (hunk 2 above)
Comment (bhalevy) [reply tools/scylla-sstable.cc:2900]: But parallel for each will propagate only one of them

Comment (bhalevy) [reply tools/scylla-sstable.cc:2900]: we can log an error at both levels

Comment (denesb) [reply tools/scylla-sstable.cc:2900]: Ok, lets do that then.

Comment (bhalevy) [reply tools/scylla-sstable.cc:2904]: done in v2

File: tools/scylla-sstable.cc [hunk 3]
```diff
@@ -258,7 +258,15 @@ const std::vector<sstables::shared_sstable> load_sstables(schema_ptr schema, sst
...
         auto sst = sst_man.make_sstable(schema, dir_path.c_str(), local, ed.generation, sstables::sstable_state::normal, ed.version, ed.format);
 
-        co_await sst->load(schema->get_sharder(), sstables::sstable_open_config{.load_first_and_last_position_metadata = false});
+        try {
+            co_await sst->load(schema->get_sharder(), sstables::sstable_open_config{.load_first_and_last_position_metadata = false});
+        } catch (...) {
+            // Print each individual error here since parallel_for_each
+            // will propagate only one of them up the stack.
+            auto msg = fmt::format("Could not load SSTable: {}", sst->get_filename());
+            fmt::print(std::cerr, "{}: {}", msg, std::current_exception());
```
Comment (denesb) [inline tools/scylla-sstable.cc:267]: You need to add a new-line because `fmt::print()` doesn't add one.

**Review by denesb — APPROVED** [review]
Looks good, one small comment.

File: tools/scylla-sstable.cc (hunk 3 above)
Comment (bhalevy) [reply tools/scylla-sstable.cc:267]: oh, right :-/

**Comment by bhalevy** [comment]
[8c6b972](https://github.com/scylladb/scylladb/commit/8c6b972e287151c6aff5f6111ba24c67b760184d):
- added `\n` to `fmt::print` format string

**Pull request closed without merging**